Cesium.CodeGen/Ir/Expressions/IndirectFunctionCallExpression.cs

```diff
@@ -0,0 +1,97 @@
+using Cesium.CodeGen.Contexts;
+using Cesium.CodeGen.Contexts.Meta;
+using Cesium.CodeGen.Extensions;
+using Cesium.CodeGen.Ir.Types;
+using Cesium.Core;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
+
+namespace Cesium.CodeGen.Ir.Expressions;
+
+internal class IndirectFunctionCallExpression : IExpression
+{
+    private readonly IdentifierExpression _function;
+    private readonly IReadOnlyList<IExpression> _arguments;
+    private readonly IExpression _callee;
+    private readonly FunctionType _calleeType;
+
+    public IndirectFunctionCallExpression(IdentifierExpression function, IExpression callee, FunctionType calleeType, IReadOnlyList<IExpression> arguments)
+    {
+        _function = function;
+        _arguments = arguments;
+        _callee = callee;
+        _calleeType = calleeType;
+    }
+
+    public IExpression Lower(IDeclarationScope scope) => this;
+
+    public void EmitTo(IEmitScope scope)
+    {
+        if (_calleeType.Parameters?.IsVarArg == true)
+            throw new CompilationException("Vararg is not supported in function pointers");
+
+        if (_callee == null)
+            throw new AssertException("Should be lowered");
+
+        VariableDefinition? varArgBuffer = null;
+        var explicitParametersCount = _calleeType.Parameters?.Parameters.Count ?? 0;
+        var varArgParametersCount = _arguments.Count - explicitParametersCount;
+        if (_calleeType.Parameters?.IsVarArg == true)
+        {
+            // TODO: See https://github.com/ForNeVeR/Cesium/issues/285
+            // Using sparse population of the parameters on the stack. 8 bytes should be enough for anybody.
+            // Also we need perform localloc on empty stack, so we will use local variable to save vararg buffer to temporary variable.
+            if (varArgParametersCount == 0)
+            {
+                scope.AddInstruction(OpCodes.Ldnull);
+            }
+            else
```

Comment:
I don't like this copy-paste from the `FunctionCallExpression`. Let's extract it somewhere, or make them lower into each other, or whatever?

Reply:
I've extracted that into a base class